specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-04-01/managedClusters.json

```diff
@@ -1532,6 +1532,24 @@
       ],
       "description": "Profile for Windows VMs in the container service cluster."
     },
+    "ManagedClusterWindowsProfile": {
+      "properties": {
+        "adminUsername": {
+          "type": "string",
+          "description": "The administrator username to use for Windows VMs.",
+          "pattern": "^[a-zA-Z0-9]+([._]?[a-zA-Z0-9]+)*$"
+        },
+        "adminPassword": {
+          "type": "string",
+          "description": "The administrator password to use for Windows VMs.",
+          "pattern": "^(?=.*[a-z])(?=.*[A-Z])(?=.*[!@#$%\\^&\\*\\(\\)])[a-zA-Z\\d!@#$%\\^&\\*\\(\\)]{12,123}$"
+        }
+      },
+      "required": [
+        "adminUsername"
```

Comment:
Can we have an empty password situation ? Isn't password also required ?

Reply:
I can't make password required because we do NOT return password in the response. Just like serviceprincipalprofile, in which we don't return the SP secret.